weaviate/client.py

```diff
@@ -54,6 +54,7 @@ def __init__(
         proxies: Union[dict, str, None] = None,
         trust_env: bool = False,
         additional_headers: Optional[dict] = None,
+        wait_for_weaviate: Optional[int] = 30,
```

Comment:
@StefanBogdan Is there any reason against having this enabled by default?

Reply:
There are 2 cases that I can think of when this functionality will kick off:
1. Weaviate is not yet ready because it is starting up. In this case it will wait until it is ready or until the time runs out. (The case we aim for)
2. The URL is wrong. This case will just ping the endpoint and will fail every time until it fails.

I think we should set a lower wait time (5 seconds) and in case it doe not start up we should throw a new Weaviate error `WeaviateStartUpError` from the caught one, with the message (or similar):
```
Weaviate did not start up in {wait_period} seconds. Either the Weaviate URL {url} is wrong or Weaivate did not start up in the interval {wait_period}.
```

Also the name `wait_for_weaviate` sounds more like a `bool` argument. Maybe something like `startup_period`/`startup_time` would be better? Or something like this.

This actually checks if Weaviate's state is `ready` not that it is `live` (currently it is the same endpoint). When weaviate is not ready it is actually possible to call the `/nodes` API. I am not sure if this is possible to do with the clients because we need the Server version in the client, so the client kind of needs Weaviate to be Ready for it to work properly. I can see how a lower default value would be fine.